pennylane/grouping/pauli_group.py

```diff
@@ -0,0 +1,273 @@
+
+    while element_idx < 4 ** n_qubits:
+        binary_string = format(element_idx, f"#0{2*n_qubits+2}b")[2:]
+        binary_vector = [float(b) for b in binary_string]
+        yield binary_to_pauli(binary_vector, wire_map=wire_map)
+        element_idx += 1
+
+
+def pauli_group(n_qubits, wire_map=None):
+    """Constructs the n-qubit Pauli group.
+
+    This function constructs and returns the complete :math:`n`-qubit Pauli
+    group (without global phases) on the desired set of wires. This function
+    differs from ``pauli_group_generator`` in that returns the group in full
+    rather than one element at a time.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in the Pauli
+            word as keys, and unique integer labels as their values. If no wire map is
+            provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        (list[qml.Operation]): The full n-qubit Pauli group.
+
+    **Example**
+
+    Construction of the Pauli group can be done as follows:
+
+    .. code-block:: python
+
+        from pennylane.grouping.pauli_group import pauli_group
+
+        n_qubits = 3
+        pg = pauli_group(n_qubits)
+    """
+    if not isinstance(n_qubits, int):
+        raise TypeError("Must specify an integer number of qubits construct the Pauli group.")
+
+    if n_qubits <= 0:
+        raise ValueError("Number of qubits must be at least 1 to construct Pauli group.")
+
+    return list(pauli_group_generator(n_qubits, wire_map=wire_map))
+
+
+def pauli_mult(pauli_1, pauli_2, wire_map=None):
+    """Multiply two Pauli words together.
+
+    Two Pauli operations can be multiplied together by taking the additive
+    OR of their binary symplectic representations.
+
+    Args:
+        pauli_1 (qml.Operation): A Pauli word.
+        pauli_2 (qml.Operation): A Pauli word to multiply with the first one.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in the Pauli
+            word as keys, and unique integer labels as their values. If no wire map is
+            provided, the map will be constructed from the set of wires acted on
+            by the input Pauli words.
+
+    Returns:
+        (qml.Operation): The product of pauli_1 and pauli_2 as a Pauli word
+            (ignoring the global phase).
+
+    **Example**
+
+    This function enables multiplication of Pauli group elements at the level of
+    Pauli words, rather than matrices. For example,
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from pennylane.grouping.pauli_group import pauli_mult
+
+        pauli_1 = qml.PauliX(0) @ qml.PauliZ(1)
+        pauli_2 = qml.PauliY(0) @ qml.PauliZ(1)
+
+        product = pauli_mult(pauli_1, pauli_2)
+        print(product)
+
+    will yield ``qml.PauliZ(0)``.
```

Comment:
Where it makes sense, it's always preferable to use pycon syntax for docstring examples (this follows the convention the NumPy docs use)
```suggestion
    >>> from pennylane.grouping.pauli_group import pauli_mult
    >>> pauli_1 = qml.PauliX(0) @ qml.PauliZ(1)
    >>> pauli_2 = qml.PauliY(0) @ qml.PauliZ(1)
    >>> product = pauli_mult(pauli_1, pauli_2)
    >>> print(product)
    PauliZ(wires=[0])
```